### ISSUE TITLE
test_convolution_format_any failed

### ISSUE THREAD

**Issue by chtlp** [01org/mkl-dnn#23]
Hi, I just installed MKL-DNN on KNL but the `test_convolution_format_any` didn't pass, any ideas what went wrong?

```
(py27) linpengt@knl-desk:~/workspace/cgt/mkl-dnn/build$  CC=icc CXX=icpc cmake .. &&  make -j

(py27) linpengt@knl-desk:~/workspace/cgt/mkl-dnn/build$ OMP_NUM_THREADS=60 make -j test
Running tests...
Test project /home/linpengt/workspace/cgt/mkl-dnn/build
      Start  1: simple-net-c
 1/22 Test  #1: simple-net-c ..........................   Passed    2.92 sec
      Start  2: simple-net-cpp
 2/22 Test  #2: simple-net-cpp ........................   Passed    4.56 sec
      Start  3: api-c
 3/22 Test  #3: api-c .................................   Passed    0.11 sec
      Start  4: test_c_symbols-c
 4/22 Test  #4: test_c_symbols-c ......................   Passed    0.01 sec
      Start  5: test_sum
 5/22 Test  #5: test_sum ..............................   Passed    0.17 sec
      Start  6: test_reorder
 6/22 Test  #6: test_reorder ..........................   Passed    0.86 sec
      Start  7: test_concat
 7/22 Test  #7: test_concat ...........................   Passed    0.10 sec
      Start  8: test_softmax_forward
 8/22 Test  #8: test_softmax_forward ..................   Passed    6.53 sec
      Start  9: test_relu
 9/22 Test  #9: test_relu .............................   Passed    0.72 sec
      Start 10: test_lrn_forward
10/22 Test #10: test_lrn_forward ......................   Passed    1.37 sec
      Start 11: test_lrn_backward
11/22 Test #11: test_lrn_backward .....................   Passed    2.99 sec
      Start 12: test_pooling_forward
12/22 Test #12: test_pooling_forward ..................   Passed    0.41 sec
      Start 13: test_pooling_backward
13/22 Test #13: test_pooling_backward .................   Passed    0.60 sec
      Start 14: test_batch_normalization
14/22 Test #14: test_batch_normalization ..............   Passed   11.43 sec
      Start 15: test_inner_product_forward
15/22 Test #15: test_inner_product_forward ............   Passed    0.26 sec
      Start 16: test_inner_product_backward_data
16/22 Test #16: test_inner_product_backward_data ......   Passed    0.12 sec
      Start 17: test_inner_product_backward_weights
17/22 Test #17: test_inner_product_backward_weights ...   Passed    0.11 sec
      Start 18: test_convolution_format_any
18/22 Test #18: test_convolution_format_any ...........***Failed    0.02 sec
      Start 19: test_convolution_forward

19/22 Test #19: test_convolution_forward ..............   Passed  366.04 sec
      Start 20: test_convolution_relu_forward
```

**Comment by skazakov1** [comment]
Thanks for reporting this. I've reproduced the issue and we will work on this.

**Comment by skazakov1** [comment]
That is just _test_convolution_format_any_ problem when this test still expects 8-elements data formats on KNL instead of 16-elements formats.

**Comment by chtlp** [comment]
Hi @skazakov1, just to make sure, the result is correct, but just the test is broken?

**Comment by skazakov1** [comment]
Yes, exactly. Results are correct.

**Comment by vpirogov** [comment]
Issue in tests is addressed with 3eed2c1b9ea08460cfdf2904fee11d9b32ecf7dc.
